src/engines_gpl/dflowfm/packages/dflowfm_kernel/src/dflowfm_io/unstruc_netcdf.f90

```diff
@@ -12634,20 +12667,28 @@ subroutine unc_read_net_ugrid(filename, numk_keep, numl_keep, numk_read, numl_re
 
                cycle
             end if
+            contact1d2didx(1, contactnlinks + L) = mesh1indexes(L)
+            contact1d2didx(2, contactnlinks + L) = mesh2indexes(L)
+            contactnetlinks(contactnlinks + L) = numl_last + L
 
+            kn(3, numl_last + l) = contacttype(l)
+
+            ! new 2D node coordinates
             XK(numk_last + l) = xface(mesh2indexes(l))
             YK(numk_last + l) = yface(mesh2indexes(l))
+            kn(2, numl_last + l) = numk_last + l
 
-            contact1d2didx(1, contactnlinks + L) = mesh1indexes(L)
-            contact1d2didx(2, contactnlinks + L) = mesh2indexes(L)
-            if (nodesOnBranchVertices == 1) then
-               kn(1, numl_last + l) = mesh1dUnmergedToMerged(mesh1indexes(l))
-            else
-               kn(1, numl_last + l) = mesh1indexes(l)
+            if (mesh1_topo_dim == 2) then
+               XK(numk_last + ncontacts + l) = xface(mesh1indexes(l))
```

Comment:
I understand where this code comes from (one net node used to be added for each contactlink, now it can be 2 net nodes for a single link), but it's a bit counterintuitive that you're first adding all *first* points for 2d2d link to the net node arrays, and after that the list of *second* points for 2d2d links. Could you change that both first and second points per contact link are added consecutively? It probably means changing the `numk_last = numk_last + ncontacts` update to incremental `numk_last = numk_last + 1` inside the `L` loop.
By the way, notice how confusing a lower case `do l=` loop is: the GitHub font has a not super obviously visible difference between `+ l` and `+ l` ;-)